lib/services/parser.go

```diff
@@ -232,11 +237,59 @@ func (ctx *Context) GetIdentifier(fields []string) (interface{}, error) {
 			session = ctx.Session
 		}
 		return predicate.GetFieldByTag(session, teleport.JSON, fields[1:])
+	case SSHSessionIdentifier:
+		// Do not expose the original session.Session, instead transform it into a
+		// ctxSession so the exposed fields match our desired API.
+		return predicate.GetFieldByTag(toCtxSession(ctx.SSHSession), teleport.JSON, fields[1:])
 	default:
 		return nil, trace.NotFound("%v is not defined", strings.Join(fields, "."))
 	}
 }
 
+// ctxSession represents the public contract of a session.Session, as exposed
+// to a Context rule.
+// See RFD 45:
+// https://github.com/gravitational/teleport/blob/master/rfd/0045-ssh_session-where-condition.md#replacing-parties-by-usernames.
+type ctxSession struct {
+	// Namespace is a session namespace, separating sessions from each other.
+	Namespace string `json:"namespace"`
+	// Login is a login used by all parties joining the session.
+	Login string `json:"login"`
+	// Created records the information about the time when session was created.
+	Created time.Time `json:"created"`
+	// LastActive holds the information about when the session was last active.
+	LastActive time.Time `json:"last_active"`
+	// ServerID of session.
+	ServerID string `json:"server_id"`
+	// ServerHostname of session.
+	ServerHostname string `json:"server_hostname"`
+	// ServerAddr of session.
+	ServerAddr string `json:"server_addr"`
+	// ClusterName is the name of cluster that this session belongs to.
+	ClusterName string `json:"cluster_name"`
+
+	// Parties is a list of session participants expressed as usernames.
+	// Changed from a []session.Party in the original session.Session.
+	Parties []string `json:"parties"`
```

Comment:
I think it would be more intuitive to call this `SessionUsernames` or `Usernames`. If I saw this w/o context, I would think `Parties` is of type `[]*Party`.

Reply:
I'm following the RFD here, but personally I do agree with you. `Usernames` seems a good call; [RFD 44](https://github.com/gravitational/teleport/blob/master/rfd/0044-session-where-condition.md) uses `session.participants`, which seems to fit well here too.

WDYT? I'm tempted to go with `Participants` so both session objects match.

Reply:
+1 for `Participants`

Reply:
Done.